lib/engine/game/g_1848/stock_market.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+require_relative '../../stock_market'
+
+module Engine
+  module Game
+    module G1848
+      class StockMarket < Engine::StockMarket
+        def move_down(corporation)
+          r, c = corporation.share_price.coordinates
+          r += 1 if (r + 1 < @market.size || r + 1 != Game::BOE_ROW) && share_price(r + 1, c)
```

Comment:
I don't believe Game::BOE_ROW will work. You'll need to pass the constant in via the constructor or something along those lines.

Reply:
I test all my changes. This works. Do you want me to change it to something else?

Reply:
@tobymao how is Game::BOE_ROW working?

Reply:
import order, it’s bad , can you import it explicitly?

Reply:
<img width="626" alt="image" src="https://user-images.githubusercontent.com/108803147/179144280-a5d75f25-af81-4a91-ae46-feb707b94c7e.png">

Reply:
would G1848::Game::BOE_ROW be better? Or should I define a method to read the constant?
Or make it a readable attribute and do @game.boe_row instead?

Reply:
the rubocop style guide references using double colon to access class instance constants here:
https://github.com/rubocop/ruby-style-guide#double-colons

SomeModule::SomeClass::SOME_CONST

Reply:
yes and make sure to require the fike

Reply:
game.rb requires stock_market.rb, so I don't think stock_market.rb can require game.rb - circular reference. I believe you'll need to pass BOE_ROW in as a constructor arg. 

Why pass in BOE_ROW instead of the game instance? We try to keep the game instance out of the basic classes to prevent too much game logic being stuffed into them.

Reply:
you can do circular references in ruby

Reply:
fixed